### PR TITLE
[CBRD-24000] CAS Statement Handlers created by Server-Side JDBC are not properly cleared

### PR DESCRIPTION
**Pull request by hgryoo**
http://jira.cubrid.org/browse/CBRD-24000

To clear the statement handler (T_SRV_HANDLE) created by server-side JDBC, `cas_procedure_handle_table` maintains the <key, value> structure (key representing the handler ID of the currently executing SQL, value representing handler ID of the server-side JDBC's query, respectively).

1) `current_handle_id` is the handler ID of the currently executing SQL issued by the client. it is set by `hm_set_current_srv_handle()` before calling `db_execute_and_keep_statement ()`.
2) Server-side JDBC requests are handled by libcas_main, and you can check whether the CAS routine is being called from libcas_main (server-side) or cas_main (client-side) through `tran_is_in_libcas()`.
3) When the client-side handler ID is freed by `hm_srv_handle_free()` at the any routine in `cas_function.c` or `cas_execute.c`, server-side hanlder IDs are destroyed (freed) according to the <key, value> structure described above.

For the <key, value> structure described above, std::multimap, C++ standard library is used.
please refer the link: https://en.cppreference.com/w/cpp/container/multimap

### CODE REVIEW

**Comment by beyondykk9** [comment]
"#if defined(LIBCAS_FOR_JSP)" macro is not necessary for current handle id ?

**Comment by hgryoo** [comment]
> "#if defined(LIBCAS_FOR_JSP)" macro is not necessary for current handle id ?

`LIBCAS_FOR_JSP` macro is always set for CAS. 

```
// cas/CMakeLists.xtxt
...
target_compile_definitions(cas PRIVATE LIBCAS_FOR_JSP)
...
```
I think it is the deprecated macro so Removing `#ifndef LIBCAS_FOR_JSP` seems to impact nothing anyway.

**Review by beyondykk9 — APPROVED** [review]

File: src/broker/cas_execute.c [hunk 1]
```diff
@@ -1247,7 +1247,9 @@ ux_execute (T_SRV_HANDLE * srv_handle, char flag, int max_col_size, int max_row,
     }
 #endif /* !LIBCAS_FOR_JSP && !CAS_FOR_ORACLE && !CAS_FOR_MYSQL */
 
+  hm_set_current_srv_handle (srv_handle->id);
```
Comment (shparkcubrid) [inline src/broker/cas_execute.c:1250]: Shouldn't it be added to ux_execute_batch()?

Comment (hgryoo) [reply src/broker/cas_execute.c:1250]: a server handler is not assigned for `ux_execute_batch()`.

**Review by shparkcubrid — APPROVED** [review]

**Comment by hgryoo** [comment]
@mhoh3963 Could you please review the CAS log format in cas.c?

**Comment by hgryoo** [comment]
I'm going to apply feedback by @mhoh3963 
1) remove "(server-side)" at the CAS log format
2) set the handler sequence number to 0 at the CAS log

```
21-07-09 01:02:46.482 (0) close_req_handle srv_h_id 1
21-07-09 01:02:46.482 (4914) close_req_handle srv_h_id 2 (server-side)
```
to
```
21-07-09 01:02:46.482 (0) close_req_handle srv_h_id 1
21-07-09 01:02:46.482 (0) close_req_handle srv_h_id 2
```

**Review by mhoh3963 — APPROVED** [review]

**Review by youngjinj — APPROVED** [review]

**Review by ctshim — APPROVED** [review]

**Pull request merged**